src/java.base/share/classes/java/lang/reflect/ProxyGenerator.java

```diff
@@ -724,91 +660,60 @@ private ProxyMethod(Method method, String methodFieldName) {
-            int nextSlot = 1;
-            for (int i = 0; i < parameterSlot.length; i++) {
-                parameterSlot[i] = nextSlot;
-                nextSlot += getWordsPerType(parameterTypes[i]);
-            }
-
-            mv.visitCode();
-            Label L_startBlock = new Label();
-            Label L_endBlock = new Label();
-            Label L_RuntimeHandler = new Label();
-            Label L_ThrowableHandler = new Label();
-
-            List<Class<?>> catchList = computeUniqueCatchList(exceptionTypes);
-            if (catchList.size() > 0) {
-                for (Class<?> ex : catchList) {
-                    mv.visitTryCatchBlock(L_startBlock, L_endBlock, L_RuntimeHandler,
-                            dotToSlash(ex.getName()));
-                }
-
-                mv.visitTryCatchBlock(L_startBlock, L_endBlock, L_ThrowableHandler,
-                        JL_THROWABLE);
-            }
-            mv.visitLabel(L_startBlock);
-
-            mv.visitVarInsn(ALOAD, 0);
-            mv.visitFieldInsn(GETFIELD, JLR_PROXY, handlerFieldName,
-                    LJLR_INVOCATION_HANDLER);
-            mv.visitVarInsn(ALOAD, 0);
-            mv.visitFieldInsn(GETSTATIC, dotToSlash(className), methodFieldName,
-                    LJLR_METHOD);
-
-            if (parameterTypes.length > 0) {
-                // Create an array and fill with the parameters converting primitives to wrappers
-                emitIconstInsn(mv, parameterTypes.length);
-                mv.visitTypeInsn(Opcodes.ANEWARRAY, JL_OBJECT);
-                for (int i = 0; i < parameterTypes.length; i++) {
-                    mv.visitInsn(DUP);
-                    emitIconstInsn(mv, i);
-                    codeWrapArgument(mv, parameterTypes[i], parameterSlot[i]);
-                    mv.visitInsn(Opcodes.AASTORE);
-                }
-            } else {
-                mv.visitInsn(Opcodes.ACONST_NULL);
-            }
-
-            mv.visitMethodInsn(INVOKEINTERFACE, JLR_INVOCATION_HANDLER,
-                    "invoke",
-                    "(Ljava/lang/Object;Ljava/lang/reflect/Method;" +
-                            "[Ljava/lang/Object;)Ljava/lang/Object;", true);
+        private void generateMethod(ClassBuilder clb, ClassDesc className) {
+            MethodTypeDesc desc = MethodTypeDesc.ofDescriptor(MethodType.methodType(returnType, parameterTypes).toMethodDescriptorString());
+            int accessFlags = (method.isVarArgs()) ? ACC_VARARGS | ACC_PUBLIC | ACC_FINAL : ACC_PUBLIC | ACC_FINAL;
+            var catchList = computeUniqueCatchList(exceptionTypes);
+            clb.withMethod(method.getName(), desc, accessFlags, mb -> {
+                ConstantPoolBuilder cpb = mb.constantPool();
+                List<ClassEntry> exceptionClassEntries = typeNames(Arrays.asList(exceptionTypes))
+                        .stream()
+                        .map(cpb::classEntry)
+                        .toList();
```

Comment:
```suggestion
                List<ClassEntry> exceptionClassEntries = Arrays.asList(exceptionTypes)
                        .stream()
                        .map(ProxyGenerator::toClassDesc)
                        .map(cpb::classEntry)
                        .toList();
```

`typeNames` streams over the given types.   This can be simplified by mapping `Class` to `ClassDesc` in place here.   I would also suggest to add  the `toClassDesc` helper method.